src/geowombat/core/util.py

```diff
@@ -487,70 +527,65 @@ def sample_feature(df_row, id_column, df_columns, crs, res, all_touched, meta, f
         return gpd.GeoDataFrame([])
 
     fid = df_row[id_column]
-
     other_cols = [col for col in df_columns if col not in [id_column, 'geometry']]
 
     if not isinstance(feature_array, np.ndarray):
-
         # "Rasterize" the geometry into a NumPy array
-        feature_array = features.rasterize([geom],
-                                           out_shape=geom_info.shape,
-                                           fill=0,
-                                           out=None,
-                                           transform=geom_info.affine,
-                                           all_touched=all_touched,
-                                           default_value=1,
-                                           dtype='int32')
+        feature_array = features.rasterize(
+            [geom],
+            out_shape=geom_info.shape,
+            fill=0,
+            out=None,
+            transform=geom_info.affine,
+            all_touched=all_touched,
+            default_value=1,
+            dtype='int64',
+        )
 
     # Get the indices of the feature's envelope
     valid_samples = np.where(feature_array == 1)
-
-    # Convert the indices to map indices
-    y_samples = valid_samples[0] + int(round(abs(meta.top - geom_info.top)) / res[1])
-    x_samples = valid_samples[1] + int(round(abs(geom_info.left - meta.left)) / res[0])
-
-    # Convert the map indices to map coordinates
-    x_coords, y_coords = meta.affine * (x_samples, y_samples)
+    # Get geometry indices
+    x_samples = valid_samples[1]
+    y_samples = valid_samples[0]
+    # Convert the geometry indices to geometry map coordinates
+    x_coords, y_coords = geom_info.affine * (x_samples, y_samples)

```

Comment:
This is the fix, part 1. The previous affine was the image affine and the replacement is the geometry affine.